agents/agent_similarity.py

```diff
@@ -64,8 +73,12 @@ def find_closest_agent(self, purpose_embedding: np.ndarray) -> Tuple[Optional[Ag
 
         try:
             for agent in self.agents:
-                agent_embedding = self.get_embedding(agent.purpose)
-                similarity = cosine_similarity([agent_embedding], [purpose_embedding])[0][0]
+                if agent.embedding_purpose is None:
+                   agent.embedding_purpose = self.get_embedding(agent.purpose)
+                else:
+                   print("\nUsing embedding_purpose variable instead of api lookup")
```

Comment:
Same here, how about we introduce a debug output?

Reply:
same here.  I just removed the print statement.  If you think this needs to be logged, we can add it back in.